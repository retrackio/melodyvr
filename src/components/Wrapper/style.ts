import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  html {
    font-size: 62.5%;
  }
`;

export const Header = styled.header`
  padding: 20px;
  background: #000;
  color: #fff;
  font-family: 'Proza Libre';
  font-weight: 600;
  font-size: 1.4rem;
  display: flex;
  align-items: center;
`;

export const HeaderTitle = styled.h1`
  font-size: 2.2rem;
  flex-grow: 1;
  margin-right: 10px;
  & svg {
    height: 30px;
  }
`;

export const MainContainer = styled.main`
  max-width: 1280px;
  margin: 0 auto;
  padding: 5px;
  background: #fff;
  color: #000;
  font-family: 'Open Sans';
  font-weight: 400;
`;
