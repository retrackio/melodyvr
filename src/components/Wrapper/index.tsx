import React from 'react';
import { Reset } from 'styled-reset';
import Link from 'next/link';
import Logo from '../Logo';
import * as Styled from './style';

interface Props {
  children: React.ReactNode;
  toolbar?: React.ReactNode;
}

const Wrapper = ({ children, toolbar }: Props) => (
  <>
    <Reset />
    <Styled.GlobalStyle />
    <Styled.Header>
      <Styled.HeaderTitle>
        <Link href="/">
          <a>
            <Logo />
          </a>
        </Link>
      </Styled.HeaderTitle>
      {toolbar}
    </Styled.Header>
    <Styled.MainContainer>{children}</Styled.MainContainer>
  </>
);

export default Wrapper;
