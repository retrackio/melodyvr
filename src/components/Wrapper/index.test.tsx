import React from 'react';
import { shallow } from 'enzyme';
import Wrapper from './';

describe('<Wrapper />', () => {
  it('is rendered correctly without a toolbar', () => {
    const wrapper = shallow(<Wrapper>some content</Wrapper>);
    expect(wrapper).toMatchSnapshot();
  });

  it('is rendered correctly with a toolbar', () => {
    const wrapper = shallow(
      <Wrapper toolbar={<div>I am the toolbar</div>}>some content</Wrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
