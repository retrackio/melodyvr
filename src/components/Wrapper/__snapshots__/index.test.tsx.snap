// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> is rendered correctly with a toolbar 1`] = `
<Fragment>
  <Memo(GlobalStyleComponent) />
  <Memo(GlobalStyleComponent) />
  <style__Header>
    <style__HeaderTitle>
      <Link
        href="/"
      >
        <a>
          <Logo />
        </a>
      </Link>
    </style__HeaderTitle>
    <div>
      I am the toolbar
    </div>
  </style__Header>
  <style__MainContainer>
    some content
  </style__MainContainer>
</Fragment>
`;

exports[`<Wrapper /> is rendered correctly without a toolbar 1`] = `
<Fragment>
  <Memo(GlobalStyleComponent) />
  <Memo(GlobalStyleComponent) />
  <style__Header>
    <style__HeaderTitle>
      <Link
        href="/"
      >
        <a>
          <Logo />
        </a>
      </Link>
    </style__HeaderTitle>
  </style__Header>
  <style__MainContainer>
    some content
  </style__MainContainer>
</Fragment>
`;
